ral-facilities/scigateway
# Redirect to plugin homepage #678

## Description
Adding support for a new universal homepage to replace the basic SciGateway page if one is set. This is configured by a new field in public/settings.json:

`"homepageUrl": "/path/to/homepage"`

The homepage can be any registered route setup by a plugin in its own settings file, e.g. public/datagateway-dataview-settings.json. Ensure the "homepageUrl" in settings.json matches the path to this route exactly.

On website load, the new homepage is displayed first (also overriding any "startUrl" currently configured. This may be changed in a later commit). Any attempts to navigate to /, such as clicking on the SciGateway logo in the main app bar or through redirects, will redirect to the new homepage.

## Testing instructions
Best test case right now is by using the DataGateway homepage. This is currently in development so should be tested using the [customisable-homepage-#431](https://github.com/ral-facilities/datagateway/tree/customisable-homepage-%23431) branch on DataGateway. 
The homepage is currently stored in datagateway-common and imported into datagateway-dataview. So the minimum required plugin to load is datagateway-dataview. [Follow the instructions on the related PR to get started.](https://github.com/ral-facilities/datagateway/pull/662)

- [x] Check that the new homepage loads when the website is first navigated to
- [x] Check that navigating to / in the address bar redirects to the new homepage
- [ ] Check that clicking the SciGateway logo on the main app bar redirects to the new homepage
- [x] Check that the new homepage is navigated to after logging in
- [x] Check that setting any other valid registered route as the new homepage passes the same tests as described above
- [x] Check that setting an invalid route as the new homepage displays a 404 page
- [x] Check that a link to the homepage does NOT display in the navigation drawer

## Agile board tracking
connect to [DataGateway PR #662](https://github.com/ral-facilities/datagateway/pull/662) and [DataGateway Issue #431](https://github.com/ral-facilities/datagateway/issues/431)
closes #678
